core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerOutputPathResolver.java

```diff
@@ -192,19 +227,35 @@ protected boolean isValidOutputPath( @NonNull String path ) {
    */
   protected String getJobName() {
     return StringUtils.isNotBlank( getFilename() )
-      ? FilenameUtils.getPathNoEndSeparator( getFilename() ) // remove "/" + <jobName> + ".*" file pattern text
+      ? stripWildcardExtension( getFilename() )
       : "<?>";
   }
 
+  private static String stripWildcardExtension( String fileName ) {
+    if ( fileName != null && fileName.endsWith( WILDCARD_EXTENSION ) ) {
+      return fileName.substring( 0, fileName.length() - WILDCARD_EXTENSION.length() );
+    }
+
+    return fileName;
+  }
+
   private String getUserSettingOutputPath() {
     try {
-      IUserSetting userSetting = getUserSettingService().getUserSetting( DEFAULT_SETTING_KEY, null );
-      if ( userSetting != null && StringUtils.isNotBlank( userSetting.getSettingValue() ) ) {
-        return userSetting.getSettingValue();
+      // TODO: This is currently returning null. May be due to the service not being registered using
+      //  `<pen:publish as-type="INTERFACES" />`, even though that isn't impeding the counterpart
+      //  `resources/SchedulerOutputPathResolver` from resolving it. Tried publishing, but then Spring would fail due to
+      //  there not being a current Spring Session context when this code runs (when a schedule is run).
```

Comment:
The fix made in this PR was only of not logging a needless NPE.
However, the underlying issue, that the user settings service is not resolvable from this class is still there.

Reply:
This is registered in https://hv-eng.atlassian.net/browse/BISERVER-15091.